### ISSUE TITLE
request - modify and add PVR views.

### ISSUE THREAD

**Issue by loggio** [jurialmunkey/skin.arctic.zephyr.2#31]
Hi Jurialmunkey,

Not sure how this skin slipped under my radar but i've only just discovered it... Hats off to you once again for a brilliant skin.

A couple of things that would be great to see addressed are to do with LiveTV views. 

**MyPVRChannels.xml**

Can i request that we have a simple ''list'' view for the pvr channel list that removes the plot description above the channel list and just shows a whole page of channels?

Although it looks awesome, the current PVR view only allows the user to see a total of 4 channels in the list at any given time... and when you consider some people have PVR setups with nearly 100 channels, it doesn't provide much of a viewing experience.

I've been using Pellucid for about 8 months as my main skin and one thing i absolutely love is the incredibly simple PVR channel list. No plot descriptions, just basic "Now-Next" type list. 
*see Pellucid skin as example.

**DialogPVRChannelsOSD.xml**

Can we have channel logos in the channel list ? they're a great visual reference when trying to quickly locate a channel and without them it gets frustrating to have to read until you find the channel you are looking for. 

There is a lot of dead space in the Logo / Cancel area to the right of this list, i feel that this space could be removed, the list could be widened and the channel logos could be added to the list instead of off to the side like they are. Basically replicate what the MyPVRChannel.xml list looks like.

**DialogSeekBar.xml**

Currently in PVR, the seekbar does not show the timeshift buffer. Can we have the bar show how much timeshift seek is available to the user? i can provide more info on this if needed.

Cheers and keep up the good work.

**Comment by jurialmunkey** [comment]
All of these are actually already on my todo list! So that's a yes to all.

**Comment by loggio** [comment]
Great! Thanks for getting back to me.

I'll report back if I find any issues.

Have a lovely Easter.

**Comment by jurialmunkey** [comment]
Just a quick note to say that I've made significant updates to DialogPVRChannelsOSD (and the rest of the PVR osd in general) - so I think that item can be said to be done now. 

Updates to MyPVRChannels and Seekbar will be coming sometime during the week.

**Comment by loggio** [comment]
Great I'll check it out very soon. One other thing while you're at it and before I forget... None of the pvr views show a "currently recording" status icon. So if you click the record button and forget about it, there's no indication that a channel is actually recording in any of the views.

**Comment by loggio** [comment]
Just tested this.. Not sure if Im sold on the osd channel list as I like to see a whole list with minimal need to scroll. Scrolling to reveal one at a time is not my cup of tea. But I don't want to sway your artistic vision at all so don't let that opinion get in your way.

I like the info dialog :-) 

However, I think everything is too transparent with a moving image in the background. and a lot of news/sports programmes have footer banners that blend too much with the dialog making it very hard to read at times. See photos.

 

![54BF61E5-FD70-4380-B6FB-2F4DB206419D](https://user-images.githubusercontent.com/16231443/59631262-be8df700-918a-11e9-8c71-2cb97a3500c1.jpeg)
![FF7B69AA-9FDF-42CA-8082-06D04E8A22AA](https://user-images.githubusercontent.com/16231443/59631263-be8df700-918a-11e9-836f-62a2863b9e49.jpeg)
![0A86F65D-6E83-41DB-9C2E-A7CDCC98FD0D](https://user-images.githubusercontent.com/16231443/59631265-be8df700-918a-11e9-93a2-e8824d46de14.jpeg)
![854F0AA7-7EAE-46F9-BA35-367F775FB918](https://user-images.githubusercontent.com/16231443/59631266-bf268d80-918a-11e9-9695-132f21a4f597.jpeg)
![4B892A31-34CA-4F88-AE24-BE95A726779B](https://user-images.githubusercontent.com/16231443/59631267-bf268d80-918a-11e9-9ccb-73685d06f279.jpeg)
![A0314FB1-42DE-4CD0-87DC-A7587203E40E](https://user-images.githubusercontent.com/16231443/59631268-bf268d80-918a-11e9-84ca-a4816c6eeea5.jpeg)

**Comment by jurialmunkey** [comment]
Yeah after using it for a few days, I definitely agree about the transparency. I'll bump up the opacity a bit in the next update.

Also I implemented a full channel list osd view:
https://i.imgur.com/wZvJ4c1.png
https://i.imgur.com/WsqiaD3.png

The single line switcher is still there by pressing up/down from osd controls and this new full channel list is shown for the osd channels list. 

Also, as a fellow Aussie I think you would really appreciate Matt Huisman's add-ons if you don't already know about them. I'm using his Foxtel Go, freeview IPTV channels, and IPTV merge add-ons to join my foxtel subscription with freeview in the kodi epg. I finally don't need to switch to another device to watch tv! https://www.matthuisman.nz/

**Comment by loggio** [comment]
Oh great I'll take it for a test run.

No I actually haven't heard of his addons. I've been using pvr since its early "unofficial" inception with kodi. It's come a long way, but I've always just used a tvheadend server running on a rpi with a hd home run network tuner. As i only watch free to air and don't have foxtel this has suited me just fine.

But since I love to try new things I'll give it a go. Thanks for the info. Where are you in Australia if you don't mind me asking?

**Comment by jurialmunkey** [comment]
If you use tvheadend you might be interested in his iptv of freeview through tvheadend. Really useful if you can't get reception on some channels. 
https://www.matthuisman.nz/2019/02/libreelec-tvheadend-iptv-freeview-au.html

This is how I'm getting freeview now. I used to use a Sony PlayTV with tvheadend, but the last two places I've lived haven't had an external antenna. I'm in Sydney in a supposedly good reception area but reception has been practically non-existent with an indoor antenna - was the main reason I got foxtel (plus it was cheap in a bundle).

**Comment by jurialmunkey** [comment]
* Added simple list for MyPVRChannels. Will show 8 rows (change to it like a normal viewtype).

Only thing left to do on this list now is figure out how the timeshift stuff works.

**Comment by loggio** [comment]
Awesome, just tried it out. looks good, nice and simple and plenty of channels visible in the list now.
The progress bar <info> lines for timeshift buffer that you'll probably need to use are 

- PVR.EpgEventProgress
- PVR.TimeshiftProgressBufferStart 
- PVR.TimeshiftProgressBufferEnd

Also, don't forget to add 'currently recording' status icons/indicators on channels that are currently recording in the channel list and epg guide. EPG guide will also need 'timer' icon/indicators on any listings that are scheduled for a recording.

and there is one thing i notice before i forget....

The OSD dialog for live tv, when you change a channel pressing up/down the seekbar flashes to the default for a split second before reverting back to the new liveTV style seekbar. It's very quick and not sure if it's fixable... might be a limitation of kodi... But you might just need to add a visibility condition for the default seekbar to prevent this. Not sure.

**Comment by loggio** [comment]
@jurialmunkey

Regarding the issue mentioned above. this happens on my NVIDIA shield (android TV) device using TVHeadend as a backend. I have not tested this on my RPi's or windows/Linux machines yet, but I know what is causing it. 

Turning off "skin.HasSetting(AutoMusicVis)" fixed this issue. So the issue is with the music visualisation triggering the music seekbar for a split second on channel start/change. I'm gussing Kodi is rendering the channel as an audio stream prior to flagging it as a PVR stream triggering the audio seekbar/visualisation briefly. A simple visibility condition on the music seekbar might fix this.

**Comment by jurialmunkey** [comment]
Hmm this is an odd issue. I have noticed something similar/potentially-related where very occasionally kodi has dumped me into the music vis rather than fullscreenvideo window and has just stayed there. I haven't been able to get it to happen lately though, but it sounds like the same thing (PVR registering the stream as a radio channel rather than a tv channel because audio starts first).

I'm not sure if I can use a visible condition on music seekbar because it might mess with radio section of pvr.

I don't usually use the AutoMusicVis setting, so I might turn it on for a while and see if I can recreate the issue.

**Comment by loggio** [comment]
Great so it's not just me then... this was happening to me the other day and has happened a few times in the past where kodi will register the channel as audio and play it as if it were a music stream. Pushing STOP would usually cause kodi to crash and I'd have to restart to fix it.
